Can't build or deploy on Ubuntu or Windows using WSL after new changes

### Describe the bug

Ubuntu & openpilot downloaded yesterday.  [Following these instructions from commaai ](https://docs.comma.ai/tools/README.html)

Basically, I can get to the point where it says:
"----   OPENPILOT SETUP DONE   ----"

but, recall, after that we have to build it, right?   (Can I just get into running it from here, without doing anything else?) 
Here's where all of the different errors come in (including a way I've found to bypass SOME of them:

`                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./tools/cabana/mainwin.h:110:54: note: passing argument to parameter 'w' here
  void drawHelpForWidget(QPainter &painter, QWidget *w);
                                                     ^
1 error generated.
scons: *** [tools/cabana/moc_mainwin.o] Error 1
1 error generated.
scons: *** [tools/cabana/detailwidget.o] Error 1
In file included from tools/cabana/mainwin.cc:1:
In file included from ./tools/cabana/mainwin.h:3:
In file included from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/QDockWidget:1:
In file included from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/qdockwidget.h:44:
In file included from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/qwidget.h:45:
/usr/include/x86_64-linux-gnu/qt5/QtCore/qobject.h:264:28: error: cannot initialize a parameter of type 'const QObject *' with an lvalue of type 'const typename QtPrivate::FunctionPointer<void (MessagesWidget::*)(const MessageId &)>::Object *' (aka 'const MessagesWidget *')
        return connectImpl(sender, reinterpret_cast<void **>(&signal),
                           ^~~~~~
tools/cabana/mainwin.cc:359:12: note: in instantiation of function template specialization 'QObject::connect<void (MessagesWidget::*)(const MessageId &), void (CenterWidget::*)(const MessageId &)>' requested here
  QObject::connect(messages_widget, &MessagesWidget::msgSelectionChanged, center_widget, &CenterWidget::setMessage);
           ^
/usr/include/x86_64-linux-gnu/qt5/QtCore/qobject.h:471:63: note: passing argument to parameter 'sender' here
    static QMetaObject::Connection connectImpl(const QObject *sender, void **signal,
                                                              ^
/usr/include/x86_64-linux-gnu/qt5/QtCore/qobject.h:265:28: error: cannot initialize a parameter of type 'const QObject *' with an lvalue of type 'const typename QtPrivate::FunctionPointer<void (MessagesWidget::*)(const MessageId &)>::Object *' (aka 'const MessagesWidget *')
                           receiver, reinterpret_cast<void **>(&slot),
                           ^~~~~~~~
tools/cabana/mainwin.cc:614:12: note: in instantiation of function template specialization 'QObject::connect<void (FindSimilarBitsDlg::*)(const MessageId &), void (MessagesWidget::*)(const MessageId &)>' requested here
  QObject::connect(dlg, &FindSimilarBitsDlg::openMessage, messages_widget, &MessagesWidget::selectMessage);
           ^
/usr/include/x86_64-linux-gnu/qt5/QtCore/qobject.h:472:63: note: passing argument to parameter 'receiver' here
                                               const QObject *receiver, void **slotPtr,
                                                              ^
/usr/include/x86_64-linux-gnu/qt5/QtCore/qobject.h:265:28: error: cannot initialize a parameter of type 'const QObject *' with an lvalue of type 'const typename QtPrivate::FunctionPointer<void (MessagesWidget::*)(const MessageId &)>::Object *' (aka 'const MessagesWidget *')
                           receiver, reinterpret_cast<void **>(&slot),
                           ^~~~~~~~
tools/cabana/mainwin.cc:620:12: note: in instantiation of function template specialization 'QObject::connect<void (FindSignalDlg::*)(const MessageId &), void (MessagesWidget::*)(const MessageId &)>' requested here
  QObject::connect(dlg, &FindSignalDlg::openMessage, messages_widget, &MessagesWidget::selectMessage);
           ^
/usr/include/x86_64-linux-gnu/qt5/QtCore/qobject.h:472:63: note: passing argument to parameter 'receiver' here
                                               const QObject *receiver, void **slotPtr,
                                                              ^
6 errors generated.
scons: *** [tools/cabana/mainwin.o] Error 1
scons: building terminated because of errors.
(openpilot-py3.11) michael@DESKTOP-HUKCPNE:~/openpilot$`

[scons.txt](https://github.com/commaai/openpilot/files/13244612/scons.txt)


Now, I was able to work around these but I have other errors.  My quick test to bypass the earlier errors was just to run openpilot without doing the setup.  I'm not sure if this is ever an acceptable way, or if my memory of what I did is correct, but when I launched something directly, it pulled up the main window and tried to get something running but still errors.  I accidentally did not capture the entire terminal, but yes, still errors:
[partial terminal capture.txt](https://github.com/commaai/openpilot/files/13244662/partial.terminal.capture.txt)


What would be the way to go around this?  I'd love to contribute, like if we have to go through files line by line and change all of the deprecated code I'm fine with doing that.  I basically want a way forward that I can contribute to the effort :)


### OS Version

Ubuntu 20.04

### openpilot version or commit

latest

### Additional info

Same errors occured on WSL or ubuntu.  Both completely fresh installs with same version of openpilot

--- a/tools/cabana/messageswidget.h
+++ b/tools/cabana/messageswidget.h
@@ -1,6 +1,7 @@
 #pragma once
 
 #include <algorithm>
+#include <optional>
 #include <set>
 #include <utility>
 #include <vector>

